README.md

```diff
@@ -75,47 +82,57 @@ or in the case of an error:
 ```ts
 type FetchProfileResponse = {
   error: string;
-  message: string;
 };
 ```
 
-This route checks if profile information has been set for the given public key,
-verifies that the NFT is still owned by the public key, and verifies that the
-NFT has an image URL set. If the public key no longer owns the NFT, or there is
-no image URL set, no NFT is returned.
+This route checks that the given public key is attached to a profile, verifies
+that the NFT is still owned by the profile, and verifies that the NFT has an
+image URL set. If the profile no longer owns the NFT, or there is no image URL
+set, no NFT is returned.
 
 The retrieval of the image URL depends on the chain:
 
 - For Stargaze, it is fetched from the Stargaze API.
-- For Juno, it queries the `nft_info` method on the smart contract of the NFT's
-  `collectionAddress`. If `extension` is present, it checks for `image`,
-  `image_uri`, and `image_url`, in that order. If any is present, it uses that
-  one. If not, it tries to fetch the data located at the URL provided by the
-  `token_uri` field. If this response is JSON, it assumes the data conforms to
-  the [ERC721 Metadata JSON
+- For all other chains, a cw721 NFT is expected. It queries the `nft_info`
+  method on the smart contract of the NFT's `collectionAddress`. If `extension`
+  is present, it checks for `image`, `image_uri`, and `image_url`, in that
+  order. If any is present, it uses that one. If not, it tries to fetch the data
+  located at the URL provided by the `token_uri` field. If this response is
+  JSON, it assumes the data conforms to the [ERC721 Metadata JSON
```

Comment:
This effectively caches the NFT's image URI. Right? Might be worth stating that explicitly. Again, this doc is good at giving us the "what" of the route, but could be more explicit about the "why" (when we expect this would be called).

Reply:
It does not cache actually. It checks every time, since NFT metadata can change, and we verify ownership to ensure they did not transfer it elsewhere.